File: frontend/src/api/employee.ts
// frontend/src/api/employee.ts
import { apiClient } from './client'

export interface EmployeePreferences {
  id: string
  user_id: string
  notification_email: boolean
  notification_sms: boolean
  preferred_view: 'month' | 'week' | 'day'
  created_at: string
  updated_at: string
}

export interface UpdatePreferencesRequest {
  notification_email?: boolean
  notification_sms?: boolean
  preferred_view?: 'month' | 'week' | 'day'
}

export interface MyScheduleEntry {
  date: string
  shift_name: string
  shift_color: string
  start_time: string
  end_time: string
  crosses_midnight: boolean
  team_name: string | null
  position: string | null
  is_overtime: boolean
  is_trade: boolean
  notes: string | null
}

export interface LeaveBalanceSummary {
  leave_type_code: string
  leave_type_name: string
  balance_hours: number
}

export interface MyDashboardData {
  today_shift: MyScheduleEntry | null
  next_shift: MyScheduleEntry | null
  upcoming_shifts: MyScheduleEntry[]
  leave_balances: LeaveBalanceSummary[]
  pending_leave_count: number
  pending_trade_count: number
}

export const employeeApi = {
  getPreferences: () =>
    apiClient.get<EmployeePreferences>('/api/users/me/preferences'),

  updatePreferences: (body: UpdatePreferencesRequest) =>
    apiClient.patch<EmployeePreferences>('/api/users/me/preferences', body),

  getSchedule: (startDate: string, endDate: string) =>
    apiClient.get<MyScheduleEntry[]>('/api/users/me/schedule', {
      params: { start_date: startDate, end_date: endDate },
    }),

  getDashboard: () =>
    apiClient.get<MyDashboardData>('/api/users/me/dashboard'),
}

File: backend/src/models/employee.rs
use serde::{Deserialize, Serialize};
use time::OffsetDateTime;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, sqlx::FromRow)]
pub struct EmployeePreferences {
    pub id: Uuid,
    pub user_id: Uuid,
    pub notification_email: bool,
    pub notification_sms: bool,
    pub preferred_view: String,
    #[serde(with = "time::serde::rfc3339")]
    pub created_at: OffsetDateTime,
    #[serde(with = "time::serde::rfc3339")]
    pub updated_at: OffsetDateTime,
}

#[derive(Debug, Deserialize)]
pub struct UpdatePreferencesRequest {
    pub notification_email: Option<bool>,
    pub notification_sms: Option<bool>,
    pub preferred_view: Option<String>,
}

#[derive(Debug, Clone, Serialize)]
pub struct MyScheduleEntry {
    pub date: time::Date,
    pub shift_name: String,
    pub shift_color: String,
    #[serde(with = "crate::models::common::time_format")]
    pub start_time: time::Time,
    #[serde(with = "crate::models::common::time_format")]
    pub end_time: time::Time,
    pub crosses_midnight: bool,
    pub team_name: Option<String>,
    pub position: Option<String>,
    pub is_overtime: bool,
    pub is_trade: bool,
    pub notes: Option<String>,
}

#[derive(Debug, Serialize)]
pub struct MyDashboardData {
    pub today_shift: Option<MyScheduleEntry>,
    pub next_shift: Option<MyScheduleEntry>,
    pub upcoming_shifts: Vec<MyScheduleEntry>,
    pub leave_balances: Vec<LeaveBalanceSummary>,
    pub pending_leave_count: i64,
    pub pending_trade_count: i64,
}

#[derive(Debug, Serialize)]
pub struct LeaveBalanceSummary {
    pub leave_type_code: String,
    pub leave_type_name: String,
    pub balance_hours: f64,
}

#[derive(Debug, Deserialize)]
pub struct MyScheduleQuery {
    pub start_date: time::Date,
    pub end_date: time::Date,
}
